simple example in documentation does not work regarding const_iterator

Hi,

the simple example found here, https://www.boost.org/doc/libs/1_78_0/doc/html/property_tree/accessing.html
gives the error below on the full code:
```
#include <boost/property_tree/xml_parser.hpp>
#include <iostream>

using boost::property_tree::ptree;

int main() {
    ptree pt;
    pt.push_back(ptree::value_type("pi", ptree("3.14159")));
    ptree::const_iterator it = pt.find("pi");
}
```

C:\Users\tsch\source\repos\ConsoleApplication2\ConsoleApplication2.cpp(10,45): error C2440: "Initialisierung": "boost::property_tree::basic_ptree<std::string,std::string,std::less<Key>>::assoc_iterator" kann nicht in "boost::property_tree::basic_ptree<std::string,std::string,std::less<Key>>::const_iterator" konvertiert werden
1>        with
1>        [
1>            Key=std::string
1>        ]

of course, it works with 'assoc_iterator', but if this is intended, the documentation needs a change.

Thomas

--- a/doc/accessing.qbk
+++ b/doc/accessing.qbk
@@ -16,7 +16,7 @@ Property tree resembles (almost is) a standard container with value type of `pai
 
 To find the value of `pi` we might do the following:
 
- __ptree__::__ptree_const_iterator__ it = pt.__ptree_find__("pi");
+ __ptree__::__ptree_const_assoc_iterator__ it = pt.__ptree_find__("pi");
  double pi = boost::lexical_cast<double>(it->second.__ptree_data__());
 
 This looks quite cumbersome, and would be even more so if `pi` value was not stored so near the top of the tree, and we cared just a little bit more about errors. Fortunately, there is another, correct way of doing it:

